src/app/rpmostree-builtin-status.c

```diff
@@ -654,6 +742,270 @@ status_generic (RPMOSTreeSysroot *sysroot_proxy,
   return TRUE;
 }
 
+static void
+append_to_summary (GString     *summary,
+                   const char  *type,
+                   guint        n)
+{
+  if (n == 0)
+    return;
+  if (summary->len > 0)
+    g_string_append (summary, ", ");
+  g_string_append_printf (summary, "%u %s", n, type);
+}
+
+static int
+compare_sec_advisories (gconstpointer ap,
+                        gconstpointer bp)
+{
+  GVariant *a = *((GVariant**)ap);
+  GVariant *b = *((GVariant**)bp);
+
+  RpmOstreeAdvisorySeverity asev;
+  g_variant_get_child (a, 2, "u", &asev);
+
+  RpmOstreeAdvisorySeverity bsev;
+  g_variant_get_child (b, 2, "u", &bsev);
+
+  if (asev != bsev)
+    return asev - bsev;
+
+  const char *aid;
+  g_variant_get_child (a, 0, "&s", &aid);
+
+  const char *bid;
+  g_variant_get_child (b, 0, "&s", &bid);
+
+  return strcmp (aid, bid);
+}
+
+static const char*
+severity_to_str (RpmOstreeAdvisorySeverity severity)
+{
+  switch (severity)
+    {
+    case RPM_OSTREE_ADVISORY_SEVERITY_NONE:
+      return "";
+    case RPM_OSTREE_ADVISORY_SEVERITY_LOW:
+      return "Low";
+    case RPM_OSTREE_ADVISORY_SEVERITY_MODERATE:
+      return "Moderate";
+    case RPM_OSTREE_ADVISORY_SEVERITY_IMPORTANT:
+      return "Important";
+    case RPM_OSTREE_ADVISORY_SEVERITY_CRITICAL:
+      return "Critical";
+    default:
+        return "";
+    }
+}
+
+static void
+print_advisories (GVariant *advisories,
```

Comment:
Hmm.  I think the advisory stuff is a great idea, but it feels like it could be a "round 2" of automatic updates patches?  Or do you think it's really integral?